test/index.ts

```diff
@@ -233,7 +232,8 @@ function testDeepPartial() {
     Assert<IsExact<DeepPartial<Set<string>>, Set<string>>>,
     Assert<IsExact<DeepPartial<Set<number[]>>, Set<number[]>>>,
     Assert<IsExact<DeepPartial<ReadonlySet<string>>, ReadonlySet<string>>>,
-    Assert<IsExact<DeepPartial<[]>, never[]>>,
+    Assert<IsExact<DeepPartial<[]>, []>>,
```

Comment:
So now, as `[]` is inferred differently, we will get `[]` instead of `never[]` which is in general fine